src/main/java/org/apache/sling/jcr/repoinit/impl/RepositoryInitializerFactory.java

```diff
@@ -122,14 +124,58 @@ public void processRepository(final SlingRepository repo) throws Exception {
                             continue;
                         }
                         final List<Operation> ops = parser.parse(new StringReader(script));
-                        log.info("Executing {} repoinit operations", ops.size());
-                        processor.apply(s, ops);
-                        s.save();
+                        String msg = String.format("Executing %s repoinit operations", ops.size());
+                        log.info(msg);
+                        applyOperations(s,ops,msg);
                     }
                 }
             } finally {
                 s.logout();
             }
         }
     }
+
+
+    /**
+     * Apply the operations within a session, support retries
+     * @param session the JCR session to use
+     * @param ops the list of operations
+     * @param logMessage the messages to print when retry
+     * @throws Exception if the application fails despite the retry
+     */
+    private void applyOperations(Session session, List<Operation> ops, String logMessage) throws RepositoryException {
+
+        RetryableOperation retry = new RetryableOperation.Builder().withBackoffBaseMsec(1000).withMaxRetries(3).build();
```

Comment:
I think at least the maximum number of retries should be configurable. Together with logging the actual retry count as suggested below, this can help investigate problematic cases.

Reply:
That is part of the Builder[1]. All repoinit operations share the same (hardcoded) retry settings. I don't see why it should be configurable in a more granular fashion.

[1] https://github.com/apache/sling-org-apache-sling-jcr-repoinit/blob/feature/SLING-10418-retry/src/main/java/org/apache/sling/jcr/repoinit/impl/RetryableOperation.java#L136

Reply:
A hardcoded setting for the number of retries is what I'd like to avoid. What if someone thinks their operation is still failing with 3 retries but would work with 4 retries? Having the backoff duration hardcoded is probably ok, but I don't see a good reason for hardcoding the max retry count. Defaulting to 3 is certainly good.

Reply:
I can make it configurable on the OSGI level, but I still question the need to actually do that. Can't we just stick with the hardcoded value for now and see if that's good enough? If we need to be more flexible, we add that feature later on.